File: Assets/Script/Lib/Agent/AgentStepBudget.lua
-- [ts]: AgentStepBudget.ts
local ____exports = {} -- 1
function ____exports.isFinalAgentDecisionTurn(agentStepCount, maxSteps) -- 3
	return agentStepCount + 1 >= maxSteps -- 4
end -- 3
function ____exports.getRemainingAgentWorkSteps(agentStepCount, maxSteps) -- 7
	local remaining = maxSteps - agentStepCount - 1 -- 8
	return remaining > 0 and remaining or 0 -- 9
end -- 7
return ____exports -- 7

File: Assets/Script/Lib/Agent/AgentStepBudget.ts
// @preview-file off clear

export function isFinalAgentDecisionTurn(agentStepCount: number, maxSteps: number): boolean {
	return agentStepCount + 1 >= maxSteps;
}

export function getRemainingAgentWorkSteps(agentStepCount: number, maxSteps: number): number {
	const remaining = maxSteps - agentStepCount - 1;
	return remaining > 0 ? remaining : 0;
}

export function getPlainTextCompletionBudgetState(agentStepCount: number, maxSteps: number): {
	outcome: "completed" | "partial";
	budgetExhausted: boolean;
} {
	const budgetExhausted = isFinalAgentDecisionTurn(agentStepCount, maxSteps);
	return {
		outcome: budgetExhausted ? "partial" : "completed",
		budgetExhausted,
	};
}
